mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -907,29 +918,78 @@ static LogicalResult printOperation(CppEmitter &emitter, ModuleOp moduleOp) {
   return success();
 }
 
+template <class FuncOpClass>
 static LogicalResult printFunctionArgs(CppEmitter &emitter,
-                                       Operation *functionOp,
+                                       FuncOpClass functionOp,
                                        ArrayRef<Type> arguments) {
   raw_indented_ostream &os = emitter.ostream();
 
+  uint32_t index = 0;
+
   return (
       interleaveCommaWithError(arguments, os, [&](Type arg) -> LogicalResult {
```

Comment:
You can use llvm::enumerate(arguments) to get a tuple of idx, argument to avoid manual counting.